### ISSUE TITLE
Add a real-valued Fourier GP transform: the complex one over-parameterises real signals by ~2x

### ISSUE THREAD

**Issue by chrisfinlay** [epfl-radio-astro/tabascal#119]
The Fourier GP machinery in `fft_gp.py` only knows how to model a **complex** signal. Every real-valued signal built on it is therefore carried as a complex one, which is roughly twice the parameters it needs and leaves half of them doing nothing.

## What is there now

`latent_to_signal` is a full complex N-D transform (`fft_gp.py:705`):

```python
Y_ss = jnp.fft.ifftshift(Y_padded)
y_padded_ss = jnp.fft.ifftn(Y_ss, norm="forward")
```

`fftn` / `ifftn` throughout — `rfftn` / `irfftn` appear nowhere — and the latent is a full complex grid held as two real parameter arrays of that shape:

```python
"ast_k_r_base": ("n_bl", "n_k_freq_ast", "n_k_time_ast"),
"ast_k_i_base": ("n_bl", "n_k_freq_ast", "n_k_time_ast"),
```

so `2N` real parameters for `N` grid modes. For a complex signal that is correct and balanced: `2N` latent degrees of freedom against `2M` in the signal. For a **real** signal of `M` samples only `M` real degrees of freedom exist, and Hermitian symmetry `Y(-k) = conj(Y(k))` says so — a bit under half the complex grid is independent, with the DC and Nyquist bins purely real.

## Why it matters

**Roughly twice the parameters.** On the shipped `ast` test grid, 9 surviving modes are 18 real parameters; the real-valued equivalent needs about 9.

**Half of them are unconstrained.** A component that needs a real signal from a complex synthesis has to take the real part somewhere. Everything the likelihood sees is then invariant to the imaginary half of the latent, which is left to the prior alone — parameters that no data constrains, wandering under the optimiser and inflating the Laplace covariance. This is the same objection #77 made to modelling a satellite below the horizon, in a different place.

**Nothing enforces realness.** A real signal is currently a convention the consumer applies, not a property of the model. Hermitian-symmetric modes make the inverse transform real *by construction*, which is what distinguishes a genuinely real-valued model from a complex one with a constraint bolted on afterwards.

## What is wanted

A real-valued sibling of the existing pair in `fft_gp.py` — `latent_to_signal` / `signal_to_latent` and their `_init` counterparts — built on `irfftn` / `rfftn` or on explicit Hermitian packing, with the latent covering only the independent half-space.

Two things to get right, both of which fail silently:

- **The degree-of-freedom count.** The independent real parameters must equal the signal samples, with the DC and Nyquist bins real-only. Off-by-one here produces a model that still runs and still fits, just with a spurious or missing direction.
- **The variance normalisation.** Under Hermitian packing a non-self-conjugate mode stands for two grid modes and contributes twice to the signal variance. Carrying `sum(pk)` across unchanged from the complex case would put the prior width out by about `sqrt(2)`. Given #45 is making that normalisation the user-facing amplitude in Jy, this needs a test that the realised signal variance matches the configured one, not just that the shapes line up.

Both are the kind of factor this codebase has produced before and which no shape check catches.

## Why it comes first

Two consumers are already queued, and both are real-valued:

- **#118 — `GPGains` to Fourier.** Gain amplitude and phase are each real. Converting it on the complex machinery would double its parameter count on arrival and bake in the redundancy.
- **#113 — a real-valued RFI-signal component.** Already lists Hermitian-symmetric modes as a requirement; this is the shared half of that work rather than something to build twice.

So this should land before #118, and #113 should consume it rather than reimplement it.

Note the FFI-kernel typing blocker recorded in #113 (`RFI FFI kernels require matched precision ... Got (amp=float64, phase=float64)`) is a constraint on *that* consumer, not on this transform — gains does not go through the RFI vis kernels.

## Acceptance

- [ ] Real-valued transform pair in `fft_gp.py`, alongside the complex one, with the latent covering only the independent half-space
- [ ] A round-trip test: signal to latent to signal is the identity on real input
- [ ] A test that the synthesised signal is exactly real, not real to rounding
- [ ] A test that the independent real parameter count equals the signal degrees of freedom
- [ ] A test that the realised signal variance matches the configured amplitude, pinning the Hermitian weighting
- [ ] Parameter count measured against the complex path on the same grid, and the ~2x recorded

## Related

- #118 — `GPGains` to Fourier, the first consumer; this precedes it
- #113 — real-valued RFI-signal component, the second consumer
- #45 — the shared prior spec whose amplitude normalisation this interacts with
- #88, #87 — the other `fft_gp` implementation issues

**Comment by chrisfinlay** [comment]
No longer a release blocker. #129 removes `GPGains` rather than converting it, so this transform is not needed for #118 (now closed).

Still wanted for #113 (real-valued RFI-signal component), which is its remaining consumer. Leaving open, but off the release path — the `~2x` over-parameterisation argument and the two silent-failure hazards (degree-of-freedom count, and the `sqrt(2)` variance normalisation under Hermitian packing) are unchanged.
